### DISCUSSION TITLE
Save to disk reducing bitrate

### DISCUSSION THREAD

**Discussion by webfrank** [bluenviron/gortsplib#657] [Questions]
Hi,
  I was trying to record a H264 RTSP stream to file, following the example, but I would like to have any advice on how to reduce filesize reducing bitrate (re-encoding i suppose).

Is it possible to hook go-ffmpeg lib for this? Or use some other method?

Any help appreciated.

**Comment by aler9** [comment]
Hello, the aim of this library is to provide a RTSP implementation, not a media stack that allows to perform decoding or re-encoding. Examples are provided in order to display possible interactions between this library and other media formats and libraries, but there is no intention of directly supporting these scenarios.

Having said this, there are several ways to introduce encoding into your Go code:

1. use FFmpeg C bindings directly into the Go code. Start from a C example that shows how to encode something with the FFmpeg C libraries, like this:
  https://gist.github.com/sdumetz/961585ea70f82e4fb27aadf66b2c9cb2

   then convert the file into Go code by using this format (the file shows how to perform decoding, but it can be adapted for encoding by replacing C calls):
   https://github.com/bluenviron/gortsplib/blob/main/examples/client-play-format-h265-convert-to-jpeg/h265_decoder.go

2. use go-ffmpeg. There are no particular differences with respect to the previous case.

3. evaluate whether it's better to use something ready-to-use like [MediaMTX + FFmpeg](https://github.com/bluenviron/mediamtx?tab=readme-ov-file#remuxing-re-encoding-compression) - if you just want to recompress the stream and record it, this feature is provided out of the box.
